### ISSUE TITLE
Failures with multiple trenches in v0.5.0

### ISSUE THREAD

**Issue by uablrek** [Nordix/Meridio#216]
The commit #f56fece0c (tagged v0.5.0) causes failures with multiple trenches. In commit #7d84c91e961 everything works fine.

I start 3 trenches, 1'st is successful, 2'nd fails sometimes, 3'rd always fail. The noticeable problem is that one proxy does not become ready;
```
NAME                                     READY   STATUS    RESTARTS   AGE   IP          NODE     NOMINATED NODE   READINESS GATES
meridio-ipam-0                           1/1     Running   0          40m   11.0.1.20   vm-001   <none>           <none>
meridio-load-balancer-86c465b4ff-4qhbg   3/3     Running   0          40m   11.0.3.13   vm-003   <none>           <none>
meridio-load-balancer-86c465b4ff-7b25b   3/3     Running   0          40m   11.0.2.12   vm-002   <none>           <none>
meridio-nse-bd7bfdbb8-7xslc              1/1     Running   0          40m   11.0.1.18   vm-001   <none>           <none>
meridio-nsp-0                            1/1     Running   0          40m   11.0.3.15   vm-003   <none>           <none>
meridio-proxy-df4q4                      1/1     Running   0          40m   11.0.2.13   vm-002   <none>           <none>
meridio-proxy-n6mbm                      1/1     Running   0          40m   11.0.3.14   vm-003   <none>           <none>
meridio-proxy-znfmg                      0/1     Running   0          40m   11.0.1.19   vm-001   <none>           <none>
```
Log [meridio-proxy-znfmg.log.txt](https://github.com/Nordix/Meridio/files/8628868/meridio-proxy-znfmg.log.txt)

The `meridio-proxy-znfmg` has not become ready in 40m. Also note that the failing proxy is located on a node that does **not** have a load-balancer. This seem to be a pattern.

I have tried to set the timeouts added in the commit;
```
# in nsp
            - name: NSP_ENTRY_TIMEOUT
              value: 2m
# In tapa
            - name: MERIDIO_NSP_ENTRY_TIMEOUT
              value: 2m
```
but it didn't help.

**Comment by uablrek** [comment]
Different number of load-balancers;
1. Seems to work (3/3 times)
2. Fails as described in this issue
3. Proxy processes becomes ready in the 3'rd trench, but target does not get a "nsc" interface

**Comment by LionelJouin** [comment]
The proxy issue we can see in the logs seems related to NSM. Could you please add other logs (including the NSM ones). I tried to run 3 trenches, everything worked fine during the first 10 minutes, after my links were broken probably due to the NSM registry (lack of memory, I will test again with higher limit).

**Comment by uablrek** [comment]
> I tried to run 3 trenches

With >1 load-balancers?

**Comment by LionelJouin** [comment]
2 LBs, 4 targets and 1 gateway per trench
```
NAMESPACE            NAME                                                  READY   STATUS    RESTARTS   AGE   IP            NODE                 NOMINATED NODE   READINESS GATES
kube-system          coredns-64897985d-68q24                               1/1     Running   0          45m   10.244.0.2    kind-control-plane   <none>           <none>
kube-system          coredns-64897985d-cdzh4                               1/1     Running   0          45m   10.244.0.3    kind-control-plane   <none>           <none>
kube-system          etcd-kind-control-plane                               1/1     Running   0          46m   172.18.0.6    kind-control-plane   <none>           <none>
kube-system          kindnet-4szrb                                         1/1     Running   0          45m   172.18.0.4    kind-worker          <none>           <none>
kube-system          kindnet-6qj6z                                         1/1     Running   0          45m   172.18.0.9    kind-worker3         <none>           <none>
kube-system          kindnet-md7qb                                         1/1     Running   0          45m   172.18.0.3    kind-worker2         <none>           <none>
kube-system          kindnet-msvrz                                         1/1     Running   0          45m   172.18.0.6    kind-control-plane   <none>           <none>
kube-system          kube-apiserver-kind-control-plane                     1/1     Running   0          46m   172.18.0.6    kind-control-plane   <none>           <none>
kube-system          kube-controller-manager-kind-control-plane            1/1     Running   0          46m   172.18.0.6    kind-control-plane   <none>           <none>
kube-system          kube-proxy-d5lfb                                      1/1     Running   0          45m   172.18.0.3    kind-worker2         <none>           <none>
kube-system          kube-proxy-hcqzg                                      1/1     Running   0          45m   172.18.0.9    kind-worker3         <none>           <none>
kube-system          kube-proxy-j4q5t                                      1/1     Running   0          45m   172.18.0.4    kind-worker          <none>           <none>
kube-system          kube-proxy-mtm9x                                      1/1     Running   0          45m   172.18.0.6    kind-control-plane   <none>           <none>
kube-system          kube-scheduler-kind-control-plane                     1/1     Running   0          46m   172.18.0.6    kind-control-plane   <none>           <none>
local-path-storage   local-path-provisioner-5ddd94ff66-bhvdk               1/1     Running   0          45m   10.244.0.4    kind-control-plane   <none>           <none>
nsm                  admission-webhook-k8s-b56cdc65f-d6tn5                 1/1     Running   0          22m   10.244.3.2    kind-worker2         <none>           <none>
nsm                  forwarder-vpp-fgk68                                   1/1     Running   0          22m   172.18.0.4    kind-worker          <none>           <none>
nsm                  forwarder-vpp-fk9h7                                   1/1     Running   0          22m   172.18.0.3    kind-worker2         <none>           <none>
nsm                  forwarder-vpp-qcgz4                                   1/1     Running   0          22m   172.18.0.9    kind-worker3         <none>           <none>
nsm                  nsm-registry-b9f46b4bd-r8jjw                          1/1     Running   0          22m   10.244.2.2    kind-worker          <none>           <none>
nsm                  nsmgr-6gc8l                                           2/2     Running   0          22m   10.244.1.3    kind-worker3         <none>           <none>
nsm                  nsmgr-fb76q                                           2/2     Running   0          22m   10.244.3.3    kind-worker2         <none>           <none>
nsm                  nsmgr-rhbgz                                           2/2     Running   0          22m   10.244.2.3    kind-worker          <none>           <none>
red                  ipam-trench-a-0                                       1/1     Running   0          19m   10.244.3.8    kind-worker2         <none>           <none>
red                  ipam-trench-b-0                                       1/1     Running   0          17m   10.244.1.10   kind-worker3         <none>           <none>
red                  ipam-trench-c-0                                       1/1     Running   0          11m   10.244.2.20   kind-worker          <none>           <none>
red                  lb-fe-attractor-trench-a-5696c89878-fgcf4             3/3     Running   0          19m   10.244.3.5    kind-worker2         <none>           <none>
red                  lb-fe-attractor-trench-a-5696c89878-pb47p             3/3     Running   0          19m   10.244.1.4    kind-worker3         <none>           <none>
red                  lb-fe-attractor-trench-b-6b44db9b96-4fk8n             3/3     Running   0          17m   10.244.1.7    kind-worker3         <none>           <none>
red                  lb-fe-attractor-trench-b-6b44db9b96-vcsvq             3/3     Running   0          17m   10.244.3.10   kind-worker2         <none>           <none>
red                  lb-fe-attractor-trench-c-78f6686b8b-9795t             3/3     Running   0          11m   10.244.3.13   kind-worker2         <none>           <none>
red                  lb-fe-attractor-trench-c-78f6686b8b-9phb8             3/3     Running   0          11m   10.244.1.13   kind-worker3         <none>           <none>
red                  meridio-operator-controller-manager-5b9897f8c-sjrqk   1/1     Running   0          20m   10.244.2.4    kind-worker          <none>           <none>
red                  nse-vlan-attractor-trench-a-6869b876bf-l2sh8          1/1     Running   0          19m   10.244.3.4    kind-worker2         <none>           <none>
red                  nse-vlan-attractor-trench-b-55dfc8f888-hhj6h          1/1     Running   0          17m   10.244.2.10   kind-worker          <none>           <none>
red                  nse-vlan-attractor-trench-c-78ff6487c9-ncwc8          1/1     Running   0          11m   10.244.1.12   kind-worker3         <none>           <none>
red                  nsp-trench-a-0                                        1/1     Running   0          19m   10.244.2.7    kind-worker          <none>           <none>
red                  nsp-trench-b-0                                        1/1     Running   0          17m   10.244.2.13   kind-worker          <none>           <none>
red                  nsp-trench-c-0                                        1/1     Running   0          11m   10.244.2.19   kind-worker          <none>           <none>
red                  proxy-trench-a-4t2jc                                  1/1     Running   0          19m   10.244.2.6    kind-worker          <none>           <none>
red                  proxy-trench-a-g7847                                  1/1     Running   0          19m   10.244.3.6    kind-worker2         <none>           <none>
red                  proxy-trench-a-wxhc8                                  1/1     Running   0          19m   10.244.1.5    kind-worker3         <none>           <none>
red                  proxy-trench-b-6jwtw                                  1/1     Running   0          17m   10.244.2.12   kind-worker          <none>           <none>
red                  proxy-trench-b-vpv8c                                  1/1     Running   0          17m   10.244.1.9    kind-worker3         <none>           <none>
red                  proxy-trench-b-xhjht                                  1/1     Running   0          17m   10.244.3.11   kind-worker2         <none>           <none>
red                  proxy-trench-c-2pdmr                                  1/1     Running   0          11m   10.244.3.14   kind-worker2         <none>           <none>
red                  proxy-trench-c-9jmhw                                  1/1     Running   0          11m   10.244.2.18   kind-worker          <none>           <none>
red                  proxy-trench-c-hj75z                                  1/1     Running   0          11m   10.244.1.14   kind-worker3         <none>           <none>
red                  target-a-cbf4c5d64-4zjwq                              2/2     Running   0          17m   10.244.2.9    kind-worker          <none>           <none>
red                  target-a-cbf4c5d64-bz5z6                              2/2     Running   0          17m   10.244.2.8    kind-worker          <none>           <none>
red                  target-a-cbf4c5d64-gwjqn                              2/2     Running   0          17m   10.244.3.9    kind-worker2         <none>           <none>
red                  target-a-cbf4c5d64-llptk                              2/2     Running   0          17m   10.244.1.6    kind-worker3         <none>           <none>
red                  target-b-7b865dcd7b-5mc6l                             2/2     Running   0          16m   10.244.2.15   kind-worker          <none>           <none>
red                  target-b-7b865dcd7b-dp7lv                             2/2     Running   0          16m   10.244.2.14   kind-worker          <none>           <none>
red                  target-b-7b865dcd7b-tfpg2                             2/2     Running   0          16m   10.244.1.11   kind-worker3         <none>           <none>
red                  target-b-7b865dcd7b-xzzb7                             2/2     Running   0          16m   10.244.3.12   kind-worker2         <none>           <none>
red                  target-c-85fd5f859b-bhkbt                             2/2     Running   0          11m   10.244.3.15   kind-worker2         <none>           <none>
red                  target-c-85fd5f859b-jzkfh                             2/2     Running   0          11m   10.244.2.21   kind-worker          <none>           <none>
red                  target-c-85fd5f859b-tkqh2                             2/2     Running   0          11m   10.244.1.15   kind-worker3         <none>           <none>
red                  target-c-85fd5f859b-tp48j                             2/2     Running   0          11m   10.244.2.22   kind-worker          <none>           <none>
spire                spire-agent-c7cjr                                     1/1     Running   0          22m   172.18.0.3    kind-worker2         <none>           <none>
spire                spire-agent-h7wdw                                     1/1     Running   0          22m   172.18.0.4    kind-worker          <none>           <none>
spire                spire-agent-lr94m                                     1/1     Running   0          22m   172.18.0.9    kind-worker3         <none>           <none>
spire                spire-server-0                                        2/2     Running   0          22m   10.244.1.2    kind-worker3         <none>           <none>
```

**Comment by LionelJouin** [comment]
In NSMgr, Do you see error like:
```
Error returned from sdk/pkg/networkservice/common/connect/connectClient.Request: rpc error: code = Unknown desc = Error returned from sdk/pkg/networkservice/common/connect/connectClient.Request: rpc error: code = Unknown desc = network service lb-fe-trench-c.trench-c.red is not found: cannot support any of the requested mechanism
```

**Comment by LionelJouin** [comment]
It seems related to the NSM registry, I tried several times with the memory and k8s registries, It always worked with the memory one and with the k8s one, the proxy running on a different node as the LBs were not starting has you described.

Here is logs exported from kind with a failing case: [logs-k8s-registry.zip](https://github.com/Nordix/Meridio/files/8631052/logs-k8s-registry.zip)
The NSC errors are in kind-worker3: red_proxy-trench-c-8648k_0fc512bc-8fcf-4b2e-a0ab-d38a4ddc4e44

**Comment by tttgiang2510** [comment]
Is this issue fixed in Meridio 0.5.1?

**Comment by uablrek** [comment]
Yes. I think that was the reason for 0.5.1

**Comment by uablrek** [comment]
Fixed by #219
